eng/pipelines/agent-software-inventory.yml

```diff
@@ -127,15 +127,34 @@ jobs:
       Get-ChildItem "${env:ProgramFiles}" | ForEach-Object {
         "$(Host), $(OSVmImage), ProgramFiles, $($_.Name), , all"
       } > programfiles-inventory.csv
-    displayName: Program Files
+    displayName: (Windows) Program Files
     workingDirectory: $(Build.ArtifactStagingDirectory)
     condition: and(succeeded(), eq(variables.Host, 'Windows'))
 
   - pwsh: |
       Get-ChildItem "${env:ProgramFiles(x86)}" | ForEach-Object {
         "$(Host), $(OSVmImage), ProgramFilesX86, $($_.Name), , all"
       } >> programfiles-inventory.csv
-    displayName: Program Files (x86)
+    displayName: (Windows) Program Files (x86)
+    workingDirectory: $(Build.ArtifactStagingDirectory)
+    condition: and(succeeded(), eq(variables.Host, 'Windows'))
+
+  - pwsh: |
+      Get-ChildItem env: | ForEach-Object {
+        "$(Host), $(OSVmImage), Environment, $($_.Name), $($_.Value), all"
+      } >> env-inventory.csv
+    displayName: Environment Variables
+    workingDirectory: $(Build.ArtifactStagingDirectory)
```

Comment:
Isn't there a possibility of secrets leaking into the pipeline artifacts here?

Reply:
I didn't think that was possible if they weren't added to the `env:` block explicitly.

Reply:
I just always assume the shell env is likely to have some extra stuff placed in there. But if devops is keeping it sanitary, then good.

Reply:
I commented it out just to be safe. Don't need to run this very often anyway.